### PR TITLE
Use references for args inside ObjectWriter 

### PR DESCRIPTION
**Pull request by sandreenko**
+ version update.

PTAL @nattress, this PR is very small.

### CODE REVIEW

**Comment by JosephTremoulet** [comment]
Should they be `const` references?

File: lib/ObjWriter/.nuget/toolchain.osx.10.10-x64.Microsoft.DotNet.ObjectWriter.nuspec [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <package >
   <metadata>
     <id>toolchain.osx.10.10-x64.Microsoft.DotNet.ObjectWriter</id>
-    <version>1.0.17-prerelease-00002</version>
+    <version>1.0.17-prerelease-00003</version>
```
Comment (nattress) [inline lib/ObjWriter/.nuget/toolchain.osx.10.10-x64.Microsoft.DotNet.ObjectWriter.nuspec]: We added an export altering the public surface area. We should bump to `1.0.18-rerelease-00001`

**Comment by sandreenko** [comment]
> Should they be const references?

ObjectWriter gets them as values:
```
extern "C" unsigned GetArrayTypeIndex(ObjectWriter *OW,
                                      ClassTypeDescriptor ClassDescriptor,
                                      ArrayTypeDescriptor ArrayDescriptor)
```
So we own this values and can change them as we want. 
What is a value to do that?

File: lib/ObjWriter/.nuget/toolchain.osx.10.10-x64.Microsoft.DotNet.ObjectWriter.nuspec (hunk 1 above)
Comment (sandreenko) [inline lib/ObjWriter/.nuget/toolchain.osx.10.10-x64.Microsoft.DotNet.ObjectWriter.nuspec]: Fixed

**Comment by JosephTremoulet** [comment]
> What is a value to do that?

Just the standard ones for why to prefer `const` by default if the only reason you're passing things by reference is to reduce call overhead:
 - Intent of code is more clear from signature
 - Easier to reason about what state changes may occur where
 - Callers can pass rvalues

**Comment by sandreenko** [comment]
The const attributes were added.

**Pull request merged**